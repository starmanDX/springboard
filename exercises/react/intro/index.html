<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Part1</title>
</head>
<body>
    <div id="root"></div>
    <script crossorigin src="https://unpkg.com/react@16/umd/react.development.js"></script>
    <script crossorigin src="https://unpkg.com/react-dom@16/umd/react-dom.development.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/babel-standalone/6.26.0/babel.js"></script>
    <script src="FirstComponent.jsx" type="text/jsx"></script>
    <script src="NamedComponent.jsx" type="text/jsx"></script>
    <script src="App.jsx" type="text/jsx"></script>
</body>
</html>