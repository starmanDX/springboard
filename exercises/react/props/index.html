<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>JSX Part2</title>
</head>
<body>
    <h1>JSX Props Part 2 - Tweets</h1>
    <div id="root"></div>
    <script crossorigin src="https://unpkg.com/react@16/umd/react.development.js"></script>
    <script crossorigin src="https://unpkg.com/react-dom@16/umd/react-dom.development.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/babel-standalone/6.26.0/babel.js"></script>
    <script src="Tweet.jsx" type="text/jsx"></script>
    <script src="App.jsx" type="text/jsx"></script>
</body>
</html>