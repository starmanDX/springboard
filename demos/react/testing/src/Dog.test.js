import React from "react";
import { render } from "@testing-library/react";
import Dog from "./Dog";

//SMOKETEST
test("it renders without crashing", () => {
  render(<Dog name="Rusty" isAdopted={true} />);
});

//SNAPSHOT TESTS
test("it matches snapshot", () => {
    const { asFragment } = render(<Dog name="Rusty" isAdopted={true}/>);
  expect(asFragment()).toMatchSnapshot();
});

test("it matches snapshot", () => {
    const { asFragment } = render(<Dog name="Rusty" isAdopted={false}/>);
    expect(asFragment()).toMatchSnapshot();
});
