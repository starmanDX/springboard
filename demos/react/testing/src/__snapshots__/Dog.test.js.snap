// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Dog"
  >
    <h1>
      Rusty
    </h1>
    <p>
      Rusty has been adopted!
    </p>
  </div>
</DocumentFragment>
`;

exports[`it matches snapshot 2`] = `
<DocumentFragment>
  <div
    class="Dog"
  >
    <h1>
      Rusty
    </h1>
    <button>
      Adopt Me!!
    </button>
  </div>
</DocumentFragment>
`;
